### PR TITLE
feat: Add more commented documentation to ClusterRoles

### PR DESCRIPTION
**Pull request by samuel-hawker**
Signed-off-by: Samuel Hawker <samuel.hawker@ibm.com>

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

Added more informative comments for users
Cleaned up some groupings

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by katheris** [comment]
@samuel-hawker had a look through all the comments and all reads fine to me and didn't spot any spelling mistakes :)

**Comment by samuel-hawker** [comment]
Thanks Kate!

File: cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ rules:
 - apiGroups:
   - ""
   resources:
+  # The cluster operator needs to access and manage service accounts to grant applications cluster permissions
```
Comment (ppatierno) [inline cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml:11]: isn't "applictions" too much generic? could we say "Strimzi components" or "other components", we are not talking about generic (user) applications.

Comment (samuel-hawker) [reply cluster-operator/src/main/resources/cluster-roles/020-ClusterRole-strimzi-cluster-operator-role.yaml:11]: Very true, will amend.

**Pull request merged**